core/spec/helpers/base_helper_spec.rb

```diff
@@ -239,4 +239,55 @@ def link_to_tracking_html(options = {})
       end
     end
   end
+
+  describe '#meta_image_data_tag' do
+    context 'when image_url_path is present' do
+      it 'returns meta tag' do
+        allow_any_instance_of(Spree::BaseHelper).to receive(:image_url_path).and_return('image_url')
+
+        expect(meta_image_data_tag).to eq "<meta property=\"og:image\" content=\"image_url\" />"
+      end
+    end
+
+    context 'when image_url_path is absent' do
+      it 'returns meta tag' do
+        allow_any_instance_of(Spree::BaseHelper).to receive(:image_url_path).and_return(nil)
+
+        expect(meta_image_data_tag).to eq nil
+      end
+    end
+  end
+
+  describe '#image_url_path' do
+    context 'when object is not a product' do
+      let!(:taxon) { build :taxon }
+
+      it 'returns false' do
+        allow_any_instance_of(Spree::BaseHelper).to receive(:instance_variable_get).and_return(taxon)
+
+        expect(image_url_path).to eq nil
+      end
+    end
+
+    context 'when object is product' do
+      let!(:product) { build :product }
+      before do
+        allow_any_instance_of(Spree::BaseHelper).to receive(:instance_variable_get).and_return(product)
+      end
+
+      context 'and has no images attached' do
+        it 'returns spree logo url' do
+          expect(image_url_path).to eq asset_path(Spree::Config[:logo])
```

Comment:
What if the shop is selling Superman and Batman figures and the user googles "Batman" and the shop has a picture of "Superman" as it's `Spree::Config[:logo]`?

Will the user see a picture of "Superman" then if they google "Batman" in google image search?